internal/pkg/aead/aead_test.go

```diff
@@ -89,3 +93,79 @@ func TestMarshalAndUnmarshalStruct(t *testing.T) {
 		t.Fatalf("expected structs to be equal")
 	}
 }
+
+// TestCipherDataRace exercises a simple concurrency test for the MicscreantCipher.
+// In https://github.com/buzzfeed/sso/pull/75 we investigated why, on random occasion,
+// unmarshalling session states would fail, triggering users to get kicked out of
+// authenticated states. We narrowed our suspicious to a data race that we uncovered
```

Comment:
nit: s/suspicious/suspicions

Also, I don't think we should describe the miscreant library as "having a data race"; the library just doesn't make any attempt at thread-safety. It might be more accurate to phrase the situation as something like: "We eventually uncovered that `sso` implicitly assumed the underlying miscreant APIs to be thread-safe; in truth, they are only thread-compatible and require external synchronization."